File: system-test/dotnet/SystemTests/Legacy/Mod05/E2eTests/PlaceOrderNegativeBaseTest.cs
using Common;
using Driver.Port.Shop.Dtos;
using SystemTests.Commons.Constants;
using SystemTests.Legacy.Mod05.E2eTests.Base;
using Shouldly;
using Xunit;

namespace SystemTests.Legacy.Mod05.E2eTests;

public abstract class PlaceOrderNegativeBaseTest : BaseE2eTest
{
    [Theory]
    [InlineData("3.5")]
    [InlineData("lala")]
    public async Task ShouldRejectOrderWithNonIntegerQuantity(string nonIntegerQuantity)
    {
        var request = new PlaceOrderRequest
        {
            Sku = CreateUniqueSku(Defaults.SKU),
            Quantity = nonIntegerQuantity,
            Country = Defaults.COUNTRY
        };

        var result = await _shopDriver!.PlaceOrderAsync(request);

        result.ShouldBeFailure();
        var error = result.Error;
        error.Message.ShouldBe("The request contains one or more validation errors");
        error.Fields.ShouldNotBeNull();
        error.Fields!.ShouldContain(f => f.Field == "quantity" && f.Message == "Quantity must be an integer");
    }
}

File: system-test/typescript/tests/legacy/mod05/e2e/place-order-negative-base-test.ts
import { expect, type TestType } from '@playwright/test';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function runPlaceOrderNegative(test: TestType<any, any>): void {
  test('shouldRejectOrderWithNonIntegerQuantity', async ({ myShopDriver }) => {
    const result = await myShopDriver.placeOrder({ sku: 'SOME-SKU', quantity: '3.5', country: 'US' });

    expect(result.success).toBe(false);
    if (!result.success) {
      expect(result.error.message).toContain('The request contains one or more validation errors');
      const quantityError = result.error.fieldErrors.find((e: { field: string; message: string }) => e.field === 'quantity');
      expect(quantityError?.message).toBe('Quantity must be an integer');
    }
  });
}
